clang/lib/Basic/Targets.cpp

```diff
@@ -165,6 +165,9 @@ std::unique_ptr<TargetInfo> AllocateTarget(const llvm::Triple &Triple,
     case llvm::Triple::OpenBSD:
       return std::make_unique<OpenBSDTargetInfo<AArch64leTargetInfo>>(Triple,
                                                                       Opts);
+    case llvm::Triple::UEFI:
+      return std::make_unique<UEFIAArch64TargetInfo>(Triple, Opts);
```

Comment:
I'm going to just guess that the original patch made a dedicated X64 UEFI target because there weren't any other users. That's not true anymore so we should probably do what the other targets do and take a mixin for the underlying architecture.

Reply:
I'm not sure how we should make a mixin. Also, from what I can find, the call convention and some of the data layouts are different between x86 and other UEFI capable ISA's. We could add in checks, the solution I came up with is simple enough lol. But I do agree a mixin would probably be better.

Reply:
I mean you should just be able to do like `if (Target.Triple.isX86())` right? I think that's how the others do it.

Reply:
I think so or it might be better to leave the data layout to the architecture side. The call convention is simple enough that it could be in the mixin.

Reply:
I've moved it over to a mixin. I'm not sure how to test if the call convention and stuff is working as expected.

Reply:
If it doesn't break any existing tests it either wasn't tested enough or it's fine.